### ISSUE TITLE
Bug: SyntaxError when compile object with $id

### ISSUE THREAD

**Issue by joshuaavalon** [sinclairzx81/typebox#322]
## Problem

**Example**
```ts
import { Type } from "@sinclair/typebox";
import { TypeCompiler } from "@sinclair/typebox/compiler";
const schema = Type.Object(
  {},
  {
    title: "Config",
    $id: "https://github.com/sinclairzx81/typebox/config.schema.json",
    description: "Configuration",
    additionalProperties: false
  }
);
const compiledSchema = TypeCompiler.Compile(schema);
console.log(schema);
```

**Log**
```
undefined:3
function check_https://github.com/sinclairzx81/typebox/config.schema.json(value) {
                    ^

SyntaxError: Unexpected token ':'
    at Function (<anonymous>)
    at Object.Compile (C:\Users\user\project\git\node_modules\@sinclair\typebox\compiler\compiler.js:441:45)
    at file:///C:/Users/user/project/git/test.js:12:37
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)
```

It seems like you should normalize the value before using it to generate code.

## Additional Information
* Node: v18.12.1
* @sinclair/typebox: 0.25.21
* ESM module

**Comment by sinclairzx81** [comment]
@joshuaavalon Hi, thanks for reporting!

Yes, this is definitely a bug! Have applied a patch on `0.25.22` which encodes the identifier using UTC-16 character codes. The encoding scheme can be seen below with the PR to amend found https://github.com/sinclairzx81/typebox/pull/323.

```typescript
const Vector = Type.Object(
  {
    x: Type.Number(),
    y: Type.Number(),
    z: Type.Number(),
  },
  {
    $id: 'http://domain.com/schemas/Vector',
  },
)

const C = TypeCompiler.Compile(Vector)

console.log(C.Code())

// function check_http_58_47_47_domain_46_com_47_schemas_47_Vector(value) {
//   return (
//     (typeof value === 'object' && value !== null && !Array.isArray(value)) &&
//     (typeof value.x === 'number' && !isNaN(value.x)) &&
//     (typeof value.y === 'number' && !isNaN(value.y)) &&
//     (typeof value.z === 'number' && !isNaN(value.z))
//  )
// }
// return function check(value) {
//   return (
//     (check_http_58_47_47_domain_46_com_47_schemas_47_Vector(value))
//  )
// }

// ... with the identifier decodable with the following.
console.log(String.fromCharCode(58)) // :
console.log(String.fromCharCode(47)) // /
console.log(String.fromCharCode(47)) // /
console.log(String.fromCharCode(46)) // .
```
If I recall, I'd previously added a catch for non-valid identifiers as part of the `TypeGuard`, but it appears this may have been refactored out. Given it's common define URI identifiers, I think this encoding is probably a better solution overall.

Will leave this issue open for a day to give you an opportunity to test `0.25.22` your side.
Thanks again!
S

**Comment by joshuaavalon** [comment]
@sinclairzx81 I confirm it is fixed in 0.25.22.
